moldesign/integrators/lammps_integrator.py

```diff
@@ -0,0 +1,193 @@
+    print 'LAMMPS could not be imported; using remote docker container'
+    force_remote = True
+else:  # this should be configurable
+    force_remote = False  # debugging
+
+import tempfile
+import os
+from copy import deepcopy
+from itertools import islice
+
+def exports(o):
+    __all__.append(o.__name__)
+    return o
+__all__ = []
+
+
+@exports
+class LAMMPSNvt(ConstantTemperatureBase):
+
+    NAME_RESULT = "result"  # Name used for dump
+    NAME_AFFECTED_ATOMS = "affected_atoms"  # Name used for grouped atoms
+    NAME_NVT_SIM = "nvt_sim"    # Name used for nvt
+    NAME_ADDFORCE = "add_user_force"    # Name used for addforce
+    
+    # TODO: raise exception if any constraints are requested ...
+    def __init__(self, *args, **kwargs):
+        super(LAMMPSNvt, self).__init__(*args, **kwargs)
+        self._prepped = False # is the model prepped?
+        self._model = None
+
+        self.lammps_system = None
+        self.total_iter = None      # How many iterations of timesteps before end of simulation
+        self.output_iter = None     # How often we dump the positions of the atoms during the simulation
+        self.traj = None
+
+    def run(self, run_for):
+        """
+        Users won't call this directly - instead, use mol.run
+        Propagate position, momentum by a single timestep using LAMMPS NVT
+        
+        Arguments:
+            run_for (int): number of timesteps OR amount of time to run for
+            self.params.timestep (float): timestep length
+
+        """
+        self.total_iter = self.time_to_steps(run_for, self.params.timestep)
+        self.output_iter = self.time_to_steps(self.params.frame_interval, self.params.timestep)
+
+        if self.total_iter < self.output_iter:
+            raise ValueError("Run duration {0} "
+                             "can\'t be smaller than frame interval {1}".format(self.total_iter, self.output_iter))
+
+        # Prepare lammps system
+        self.prep()
+        self._configure_system()
+        lmps = self.lammps_system
+
+        predictable_filename = "dump.result"
+        dump_filepath = os.path.join(self.mol.energy_model.tmpdir, predictable_filename)
+        self.mol.energy_model.files.append(dump_filepath)
+
+        dump_command = "dump {0} all custom {1} {2} id x y z vx vy vz".format(self.NAME_RESULT,
+                                                                            self.output_iter, dump_filepath)
+
+        lmps.command(dump_command)
+        lmps.command("dump_modify {0} sort id".format(self.NAME_RESULT))
+
+        # run simulation
+        lmps.run(self.total_iter, "post no")
+
+        # Reset lammps system by undumping and unfixing
+        for dmp in lmps.dumps:
+            lmps.command("undump " + dmp.get('name'))
+
+        for fix in lmps.fixes:
+            lmps.command("unfix " + fix.get('name'))
+
+        # Create trajectory object
+        self.traj = Trajectory(self.mol, unit_system=self._model.unit_system)
```

Comment:
Don't specify `unit_system` here - it's actually for _output_, not input, so MDT will set it automatically.